### PR TITLE
main param selection form can now also submit jobs directly

### PR DESCRIPTION
**Pull request by forman**
If you review this PR, make sure you execute the notebook `notebooks/mui-demo.ipynb` in parallel which explains the concepts introduced in the many code changes.

Note there are several open issues in this PR which may individually turned into separate issues and then PRs:

image

### CODE REVIEW

File: multiply_ui/ui/job/model.py [hunk 1]
```diff
@@ -13,8 +13,8 @@
     PropertyDef("id", TypeDef(str)),
     PropertyDef("name", TypeDef(str)),
     PropertyDef("progress", TypeDef(int)),
-    PropertyDef("status", TypeDef(str)),
-    PropertyDef("tasks", TypeDef(list, item_type=TASK_TYPE)),
+    PropertyDef("status", TypeDef(str, choices=['new', 'running', 'succeeded', 'cancelling', 'cancelled'])),
```
Comment (thomasstorm) [inline multiply_ui/ui/job/model.py]: Maybe we'll need the status "failed" here?

File: notebooks/mui-demo.ipynb [hunk 2]
```diff
@@ -28,122 +30,286 @@
...
+       "<multiply_ui.ui.req.model.ProcessingRequest at 0x18715a79a90>"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
    "source": [
     "req"
```
Comment (thomasstorm) [inline notebooks/mui-demo.ipynb:181]: Error message would be nice when req is None.

File: notebooks/mui-demo.ipynb [hunk 3]
```diff
@@ -28,122 +30,286 @@
...
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 14,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div><p>Name: <br/>Time range: (datetime.datetime(2010, 1, 1, 0, 0), datetime.datetime(2019, 1, 1, 0, 0))<br/>Region box: (0.0, 0.0, 360.0, 180.0)Input types: S2_L1C</p><div><h4>Inputs</h4><table><tr><th>Type</th><th>Identifier</th></tr><tr><td>S2_L1C</td><td>iid-0</td></tr><tr><td>S2_L1C</td><td>iid-1</td></tr><tr><td>S2_L1C</td><td>iid-2</td></tr><tr><td>S2_L1C</td><td>iid-3</td></tr><tr><td>S2_L1C</td><td>iid-4</td></tr><tr><td>S2_L1C</td><td>iid-5</td></tr><tr><td>S2_L1C</td><td>iid-6</td></tr><tr><td>S2_L1C</td><td>iid-7</td></tr><tr><td>S2_L1C</td><td>iid-8</td></tr><tr><td>S2_L1C</td><td>iid-9</td></tr></table></div></div>"
```
Comment (thomasstorm) [inline notebooks/mui-demo.ipynb]: Add line break after "Region Box: (...)"

**Comment by thomasstorm** [comment]
If I run the notebook, I'll get errors after line
req.submit('my_job'), probably because of
'NameError: name 'my_job' is not defined'

**Review by thomasstorm — CHANGES_REQUESTED** [review]
See comments

File: multiply_ui/ui/__init__.py [hunk 4]
```diff
@@ -1,9 +1,3 @@
 from .mui import mui
 
-# TODO: remove following imports
-
-# noinspection PyUnresolvedReferences
-from .data import data_ui
```
Comment (TonioF) [inline multiply_ui/ui/__init__.py:6]: With these changes, Notebooks "multiply-ui-1.ipynb" and "multiply-ui-2.ipynb" have become obsolete. Please remove them, at the very least from "readme.txt"

File: multiply_ui/ui/req/form.py [hunk 5]
```diff
@@ -16,18 +19,29 @@
...
-        time.sleep(3)
-        input_identifiers = {input_type: [f'iid-{i}' for i in range(10)] for input_type in
-                             input_request.input_types}
-        processing_request_data = input_request.as_dict()
-        processing_request_data.update(dict(inputIdentifiers=input_identifiers))
-        apply_func(ProcessingRequest(processing_request_data))
-
-    fetch_inputs_func = fetch_inputs_mock if mock else fetch_inputs
+    fetch_inputs_func = fetch_inputs
+    if mock:
```
Comment (TonioF) [inline multiply_ui/ui/req/form.py:23]: I think a single "if mock" is enough.

**Review by TonioF — APPROVED** [review]
Just minor remarks.

**Comment by TonioF** [comment]
I have considered the comments from @thomasstorm and myelf. I have further moved the submit mocking to the api module, so it can be used from both from and model. @forman , please have a look at https://github.com/multiply-org/multiply-ui/pull/14/commits/db8526f42da3d3b4a3050c89b4e121373b1f452c .

**Comment by TonioF** [comment]
Or rather, look at https://github.com/multiply-org/multiply-ui/pull/14/commits/006855b7cbf2664851d138ecb676cb3add608e66

**Review by thomasstorm — APPROVED** [review]

**Pull request merged**